Add script to generate js from svg

rautesamtr opened the pull request:
This adds a script to generate the js frontend file from the svg files.

To support the script and new icon picker feature of home assistant the
svg need to have their name in the id of the svg file and the desc tag
added with possible keywords for the lookup.

Keywords can be added space separated in the svg `desc` tag.

The old compatibility part of the js file has been dropped. Support for non Icon Picker version of Home Assistant are still there.

hulkhaugen commented:
If I understand you correctly, turning the individual entries in `BHA_ICONS_MAP` into dicts and adding `"keywords": []`, will make the pack supported by the HA icon picker?

Your script needs an `id` and `desc` tag in the svg to generate the js file?

Sounds very neat ;) But wouldn't using filename instead of `id` be a cleaner way to do this? Will the script fail if there is no `desc` tag in the svg, or will it still generate an empty `"keywords": []` for the js? I'm not very good with either Python nor js, so I would never come up with this myself. Just thoughts form the top of my head.

rautesamtr commented:
> If I understand you correctly, turning the individual entries in BHA_ICONS_MAP into dicts and adding "keywords": [], will make the pack supported by the HA icon picke

This plus you have to use 
```js
window.customIcons = window.customIcons || {};
window.customIcons["bha"] = { getIcon, getIconList };
```
for the icon picker and additionaly
```js
window.customIconsets = window.customIconsets || {};
window.customIconsets["bha"] = getIcon;
```
for backwards compatibility.

As for your questions I can change the script to do that for you. The current version is the one i use with a [detached fork](https://github.com/rautesamtr/thermal_comfort_icons/) of your project for my own icon set. And the way it works makes sense for me but i understand that its easier for you to handle with your proposed changes.

rautesamtr committed:
Add script to generate js from svg

This adds a script to generate the js frontend file from the svg files.
This new script supports the new icon picker.
The script uses the filebasename (e.g. icon from icon.svg) as icon name.
To add keywords for the icon picker search add a desc tag to your svg
and fill it with space separated keywords.
e.g. <desc>light ceiling</desc>
The script will only pick up the first desc tag of the document.
If the tag is missing the keywords will be set to an empty list.
The script will also only pick up the first path as the icons path so
make sure they are all single path icons.

rautesamtr commented:
![image](https://user-images.githubusercontent.com/1131371/151605858-154d6323-df2b-4a7e-a7c7-4dac6f060513.png)
